src/chainrules.jl

```diff
@@ -0,0 +1,40 @@
+@non_differentiable validate_woodbury_arguments(A, D, S)
+
+function ChainRulesCore.rrule(
+    ::typeof(*), A::Real, B::WoodburyPDMat{T, TA, TD, TS}
+) where {T, TA, TD, TS}
+    project_A = ProjectTo(A)
+    project_B = ProjectTo(B)
+    function times_pullback(ȳ::AbstractMatrix)
+        Ȳ = unthunk(ȳ)
+        Ā = dot(Ȳ, B)
+        B̄ = A' * Ȳ
+        return (
+           NoTangent(),
+           @thunk(project_A(Ā')),
+           @thunk(project_B(B̄)),
+        )
+    end
+
+    function times_pullback(ȳ::Tangent{<:WoodburyPDMat})
+        Ȳ = unthunk(ȳ)
```

Comment:
```suggestion
    function times_pullback(Ȳ::Tangent{<:WoodburyPDMat})
```
We already know it's a tangent, not thunk. We should probably add a pullback dispatch which just unthunks. This broke type inference before, and the fix was to define the pullback methods outside, see something like 
https://github.com/JuliaDiff/ChainRules.jl/blob/e6a27db89cbb392a1f55fba7bae19caebd9fe43a/src/rulesets/LinearAlgebra/structured.jl#L107-L114